pinot-core/src/main/java/org/apache/pinot/core/transport/ServerChannels.java

```diff
@@ -151,7 +153,24 @@ private class ServerChannel {
 
     ServerChannel(ServerRoutingInstance serverRoutingInstance) {
       _serverRoutingInstance = serverRoutingInstance;
+      PooledByteBufAllocator bufAllocator = PooledByteBufAllocator.DEFAULT;
+      PooledByteBufAllocatorMetric metric = bufAllocator.metric();
+      _brokerMetrics.setOrUpdateGauge(BrokerGauge.NETTY_POOLED_USED_DIRECT_MEMORY.getGaugeName(),
+          metric.usedDirectMemory());
+      _brokerMetrics.setOrUpdateGauge(BrokerGauge.NETTY_POOLED_USED_HEAP_MEMORY.getGaugeName(),
```

Comment:
I think the reason we don't have that method is because local gauges (like the ones that need table names) requires to explicitly send the table name. Given that we use the same type for both global and local metrics, we cannot ensure that the gauge we receive is global here.

I can add some usability methods, but I would like to make this PR as simple as possible to merge it soon